qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/plot_widget/plotting_canvas/plotting_canvas_view.py

```diff
@@ -157,6 +170,10 @@ def clear_all_workspaces_from_plot(self):
         for color_queue in self._color_queue:
             color_queue.reset()
 
+        for shaded_region in self._shaded_regions:
+            shaded_region.remove()
```

Comment:
Do we have any tests to check the old shaded region / new region gets replaced? 
Especially since this fixed a bug we probably don't want to re-fix in the future

Reply:
because this view is all the interaction with matplotlib we have very limited testing. I can look what can be done, but I doubt it will be much

Reply:
But we can still mock out MPL and test we're calling remove right?
I'm trusting (well more hoping knowing MPL) it will do the right thing. But I'm more concerned we're making the correct calls to actually remove items?

Reply:
I can have a look. I have got a build ready, but there is some other stuff I need to do first.

Reply:
I think realistically this is the only bit I can test